### PR TITLE
[cc65] Issue #1438 fix

### LINKED ISSUE #1438: Odd crash in cc65, related to delayed post-counting
This one is really super odd. It happens when trying to compile this file:

https://github.com/EtchedPixels/FUZIX/blob/master/Applications/games/z1.c

apparently this worked at some point...

what happens is that it crashes with a segfault. apparently in the function CheckedSizeOf.

Now, i compiled cc65 with -O0 -g to track it down further... and the bug is gone... or different?. AAARGS

I know, its a terrible bug report. I tried reproducing it with a small code snippet but couldnt do it. Perhaps some array is overflowing? Some off by one error?

No idea how to proceed on this one

this is with pr #1435 merged locally, did not check yet if it is related

### PR DESCRIPTION
**Pull request by acqn**
Fixed #1438.

### CODE REVIEW

**Comment by oliverschmidt** [comment]
@acqn: Thanks for the quick fix :-)

@mrdudz: Do you want to create and merge a test before I merge this PR?

File: src/cc65/datatype.c [hunk 1]
```diff
@@ -1085,11 +1085,26 @@ int HasUnknownSize (const Type* T)
...
+    return (GetFuncDesc (T)->Flags & FD_VARIADIC) != 0;
+}
+
+
+
+int IsFastcallFunc (const Type* T)
+/* Return true if this is a function type or pointer to function type by
+** __fastcall__ calling convention.
+** Check fails if the type is not a function or a pointer to function.
+**/
```
Comment (oliverschmidt) [inline src/cc65/datatype.c:1101]: Please use the predominant multi-line comment style ending with `*/`.

File: src/cc65/datatype.h [hunk 2]
```diff
@@ -825,8 +825,15 @@ INLINE int IsQualCConv (const Type* T)
 
 int IsVariadicFunc (const Type* T) attribute ((const));
 /* Return true if this is a function type or pointer to function type with
-** variable parameter list
-*/
+** variable parameter list.
+** Check fails if the type is not a function or a pointer to function.
+**/
```
Comment (oliverschmidt) [inline src/cc65/datatype.h:830]: Please use the predominant multi-line comment style ending with `*/`.

File: src/cc65/datatype.h [hunk 3]
```diff
@@ -825,8 +825,15 @@ INLINE int IsQualCConv (const Type* T)
...
+
+int IsFastcallFunc (const Type* T) attribute ((const));
+/* Return true if this is a function type or pointer to function type by
+** __fastcall__ calling convention.
+** Check fails if the type is not a function or a pointer to function.
+**/
```
Comment (oliverschmidt) [inline src/cc65/datatype.h:836]: Please use the predominant multi-line comment style ending with `*/`.

**Review by oliverschmidt — CHANGES_REQUESTED** [review]

File: src/cc65/datatype.c (hunk 1 above)
Comment (acqn) [reply src/cc65/datatype.c]: Fixed.

File: src/cc65/datatype.h (hunk 2 above)
Comment (acqn) [reply src/cc65/datatype.h]: Fixed.

File: src/cc65/datatype.h (hunk 3 above)
Comment (acqn) [reply src/cc65/datatype.h]: Fixed.

**Comment by mrdudz** [comment]
Is this against head or with #1435 ?

**Review by oliverschmidt — APPROVED** [review]

**Comment by mrdudz** [comment]
It's becoming difficult to come up with a testcase.... because the compiler randomly crashes or not, apparently more so the more i strip away from that file mentioned above. really annoying =P

**Comment by mrdudz** [comment]
OK i have a testcase.... the problem with this is... it crashes the compiler, sometimes (maybe one of 10 compilation runs). i cant really add it to the testbench like this.

so merge, then i will add it to the working tests directly :)

**Pull request merged**